### PR TITLE
Updating the search doc

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/open-cluster-management/backlog/issues/7311

### CODE REVIEW

File: apis/search.json.adoc [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+=== URI scheme
+[%hardbreaks]
+__BasePath__ : /kubernetes/apis
+__Schemes__ : HTTPS
+
+
+=== Tags
+
+* search.open-cluster-management.io/v1beta1 : Create and manage persistence volumes
+
```
Comment (dockerymick) [inline apis/search.json.adoc:36]: From review with Sherin Update v1beta1 to `v1`

File: console/search.adoc [hunk 2]
```diff
@@ -1,11 +1,51 @@
 [#search-in-the-console]
 = Search in the console
 
-For {product-title}, Search provides visibility into your resources across all your clusters.
+For {product-title}, search provides visibility into your resources across all of your clusters. Search uses Redisgraph (`redisgraph-tls`), an in-memory database to store the Kubernetes resources and the resource relationships. Create a `searchCustomization` custom resource (CR) to define the storage settings for search persistence.
```
Comment (SherinV) [inline console/search.adoc]: We need `searchcustomization` lower case, not camel case.

File: console/search.adoc [hunk 3]
```diff
@@ -1,11 +1,51 @@
 [#search-in-the-console]
 = Search in the console
 
-For {product-title}, Search provides visibility into your resources across all your clusters.
+For {product-title}, search provides visibility into your resources across all of your clusters. Search uses Redisgraph (`redisgraph-tls`), an in-memory database to store the Kubernetes resources and the resource relationships. Create a `searchCustomization` custom resource (CR) to define the storage settings for search persistence. 
 
-You can type any text value in the _Search box_ and results include anything with that value from any property, such as a name or namespace. If you search for `RedHat` and you manage a large number of clusters, it might take a longer time to receive search results.
+*Note:* If the `redisgraph-tls` pod crashes, data is resent from the `search-collector` pod to rebuild the cache.
```
Comment (SherinV) [inline console/search.adoc]: Note not relevant in this context

File: console/search.adoc [hunk 4]
```diff
@@ -1,11 +1,51 @@
...
 
-For more specific search results, include the property selector in your search. You can combine related values for the property, for a more precise scope of your search. For example, search for `cluster:dev red` to receive results that match the string "red" in the `dev` cluster.
+[#search-customization]
+== Search customization
+
+Update the storage settings for search by updating the `searchCustomization` CR. Search customization is namespace-scoped and located where search is installed in the hub cluster. By default, `searchCustomization` CRD is created.
```
Comment (SherinV) [inline console/search.adoc]: We need searchcustomization lower case, not camel case.
Update the storage settings for search by creating the `searchcustomization` CR. 

> By default, `searchCustomization` CRD is created.

This need not be mentioned.

File: console/search.adoc [hunk 5]
```diff
@@ -1,11 +1,51 @@
...
-For {product-title}, Search provides visibility into your resources across all your clusters.
+For {product-title}, search provides visibility into your resources across all of your clusters. Search uses Redisgraph (`redisgraph-tls`), an in-memory database to store the Kubernetes resources and the resource relationships. Create a `searchCustomization` custom resource (CR) to define the storage settings for search persistence. 
 
-You can type any text value in the _Search box_ and results include anything with that value from any property, such as a name or namespace. If you search for `RedHat` and you manage a large number of clusters, it might take a longer time to receive search results.
+*Note:* If the `redisgraph-tls` pod crashes, data is resent from the `search-collector` pod to rebuild the cache.
 
-For more specific search results, include the property selector in your search. You can combine related values for the property, for a more precise scope of your search. For example, search for `cluster:dev red` to receive results that match the string "red" in the `dev` cluster.
+[#search-customization]
+== Search customization
+
```
Comment (SherinV) [inline console/search.adoc:8]: The default installation of ACM will be configured to persist search index. We can see the volume being used in the search-redisgraph statefulset definition with the volume name `persist`.

File: console/search.adoc [hunk 6]
```diff
@@ -1,11 +1,51 @@
...
+Update the storage settings for search by updating the `searchCustomization` CR. Search customization is namespace-scoped and located where search is installed in the hub cluster. By default, `searchCustomization` CRD is created.
+
+Your cluster must have a default storage class defined. If your cluster has a defined default storage class, the search component creates a Persistent Volume Claim (PVC) of 10G on the storage class. Verify if there is a default storage class in your cluster by running the following command: `kubectl get storageclass`.
```
Comment (SherinV) [inline console/search.adoc]: > Your cluster must have a default storage class defined.

This is not a requirement as we fallback to emptyDir by default

File: console/search.adoc [hunk 7]
```diff
@@ -1,11 +1,51 @@
...
+[#search-customization]
+== Search customization
+
+Update the storage settings for search by updating the `searchCustomization` CR. Search customization is namespace-scoped and located where search is installed in the hub cluster. By default, `searchCustomization` CRD is created.
+
+Your cluster must have a default storage class defined. If your cluster has a defined default storage class, the search component creates a Persistent Volume Claim (PVC) of 10G on the storage class. Verify if there is a default storage class in your cluster by running the following command: `kubectl get storageclass`. 
+
+If a default storage class is absent or search fails to create a PVC, search uses the `emptyDir` as a volume. You can disable use of the `emptyDir` by updating the `persistence` flag to `False` in the customization CR. Review the following scenarios when you update the `storageClass` flag and there is a failure to create the PVC:
```
Comment (SherinV) [inline console/search.adoc]: You can disable persistence by updating the `persistence` flag to `false` in the customization CR which will turn off saving search index.

File: console/search.adoc [hunk 8]
```diff
@@ -1,11 +1,51 @@
...
+
+* When the persistence flag is set to `True`:
+** If the `storageClass` is set, search uses the provided storage class.
+** If the `storageClass` is not set, search attempts to use the default storage class. If there is no default storage class, search uses the `emptyDir` volume.
```
Comment (SherinV) [inline console/search.adoc]: > If there is no default storage class, search uses the emptyDir volume.

This is not true any more when persistence flag is set to `True`

File: console/search.adoc [hunk 9]
```diff
@@ -1,11 +1,51 @@
...
+
+* When the persistence flag is set to `False`:
+** The `storageClass` or `storageSize` has no effect.
+** Search disables persistence.
+
+A status for persistence can be retrieved from the search operator (`searchoperator`). For more information, see link:https://kubernetes.io/docs/concepts/storage/volumes/#emptydir[Volumes] in the Kubernetes documentation. View the following table to view status messages for the `searchCustomization` CR:
```
Comment (SherinV) [inline console/search.adoc]: A status for persistence can be retrieved from the search operator (searchoperator) should suffice. The status should be self explanatory. We can remove the table.

File: console/search.adoc (hunk 5 above)
Comment (dockerymick) [reply console/search.adoc:8]: @SherinV Here is my incoming update: When you install {product-title-short}, the product is configured to persist search index.

Should the article "the" come before search index? 

`...is configured to persist the search index? `

**Review by SherinV — APPROVED** [review]

**Pull request merged**